tooling/src/bin/mind_benchmark.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// Here we emulate the format of the Newscatcher API, so that it's compatible with
+// our current client in the discovery engine.
+
+use std::fs::File;
+use std::{collections::HashMap, error::Error};
+
+use csv::{DeserializeRecordsIntoIter, Reader, ReaderBuilder};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct Impression {
+    _id: String,
+    _user_id: String,
+    _time: String,
+    clicks: String,
+    _news: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Article {
+    id: String,
+    _category: String,
+    _subcategory: String,
+    _title: String,
+    _snippet: String,
+    _url: String,
+}
+
+/// Reads and deserializes news data from a tsv file path and returns an iterator
+fn read_articles(path: &str) -> Result<DeserializeRecordsIntoIter<File, Article>, Box<dyn Error>> {
```

Comment:
I see, changed it to `anyhow::Error` for now. With unwrap do u mean something like https://doc.rust-lang.org/rust-by-example/error/option_unwrap.html?

Reply:
exactly, unwrapping in rust is getting the inner value of something without checking before if it's actually there. in case it's not there, the programm will usually panic. you can find unwrapping on `Result`s, `Option`s and some other types.